### PR TITLE
Connecting json data from forms in App.js to visualization

### PR DESCRIPTION
**Pull request by kaylanhua**

### CODE REVIEW

File: react-frontend-app/src/App.js [hunk 1]
```diff
@@ -12,15 +12,18 @@ import './style/App.css';
...
 import MainNavBar from './components/MainNavBar'
 import BarGraph from './components/BarGraph'
 
 function App() {
   const [surname, setSurname] = useState('');
   const [zipcode, setZip] = useState('');
-  var prediction = {};
+  const [loading, setLoading] = useState(true);
+
+  // var prediction = {};
```
Comment (chrisong88) [inline react-frontend-app/src/App.js]: Could you add a comment here explaining what information is stored in "variable"? Right now, it's a little unclear

File: react-frontend-app/src/App.js [hunk 2]
```diff
@@ -32,42 +35,85 @@ function App() {
...
   // Function that called when the submit button is pressed
-  function handleSubmit(event) {
+  async function handleSubmit(event) {
     event.preventDefault();
-    console.log('Submit was pressed');
-    console.log({surname}, {zipcode});
 
-    axios.get(`http://localhost:5000/surgeo?surname=${surname}&zipcode=${zipcode}`)
-      .then(res => console.log(res.data));
+    setVariable(await axios.get(`http://localhost:5000/surgeo?surname=${surname}&zipcode=${zipcode}`)
```
Comment (chrisong88) [inline react-frontend-app/src/App.js]: Question: is it necessary to call setVariable twice? Isn't it possible to update variable by calling:

await axios.get(`http://localhost:5000/surgeo?surname=${surname}&zipcode=${zipcode}`).then(...)

?

Comment (chrisong88) [reply react-frontend-app/src/App.js]: I might be wrong on this, so feel free to push back if I'm misinterpreting something

File: react-frontend-app/src/App.js [hunk 3]
```diff
@@ -32,42 +35,85 @@ function App() {
...
+            // name={variable.name[0]}
+            white={variable.white[0]}
+            black={variable.black[0]}
+            api={variable.api[0]}
+            hispanic={variable.hispanic[0]}
+            multiple={variable.multiple[0]}
+            native={variable.native[0]}
+          />
+          {/* <BarGraph /> */}
+        </div>
```
Comment (chrisong88) [inline react-frontend-app/src/App.js]: IMPT: It looks like the only difference between the if/else statement return values is the "Breakdown" section (ie: lines 96-108). Everything else is duplicated, which makes this redundant in terms of design. This code duplication makes the code more difficult to maintain and prone to bugs. For example, if someone wanted to update the "Zest AI Race Predictor Prototype" title, they would have to remember to make the exact same changes in both the "if" return statement and the "else" return statement. To summarize, the current set-up is really vulnerable to the two versions getting out of sync.

Can you instead factor out the "Breakdown" section into a conditionally defined Javascript statement that is conditionally set to either "Nothing here, submit your data" or "\<h3>Breakdown...etc"?

Comment (chrisong88) [reply react-frontend-app/src/App.js]: In fact, I think you could take advantage of Javascript's conditional operator to do something like this:

\<div>
    {(variable == "this is the default variable" || loading) ? \<h3>Nothing here, submit your data!\</h3> : 
         \<h3>Breakdown... etc}
\</div>

Comment (chrisong88) [reply react-frontend-app/src/App.js]: Essentially, in the above code snippet, the code within the {} is Javascript using the following format:

condition ? valueIfConditionIsTrue : valueIfConditionIsFalse

(end Javascript)

By putting the if/else within the \<div>, it precludes the need to have two return statements and makes it easier to update the code in the future! Here's some more documentation here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

File: react-frontend-app/src/components/BarGraph.js [hunk 4]
```diff
@@ -59,33 +58,61 @@ class BarGraph extends React.Component {
...
+
         var zippedData = [];
         var races = ["white", "black", "api", "native", "multiple", "hispanic"];
         var racesCaps = ["White", "Black", "Asian/PI", "Native", "Multiple", "Hispanic"];
-        for (var i=0; i<6; i++) {
+        for (var i = 0; i < 6; i++) {
             zippedData.push(parseFloat(data[races[i]].zest));
-            zippedData.push(parseFloat(data[races[i]].bisg));
+            // zippedData.push(parseFloat(data[races[i]].bisg));
+            switch (races[i]) {
```
Comment (chrisong88) [inline react-frontend-app/src/components/BarGraph.js:102]: This is so clean. Great use of switch statements.

File: react-frontend-app/src/components/Display.js [hunk 5]
```diff
@@ -1,7 +1,11 @@
 import React from 'react';
 import '../style/Display.css';
 
+<<<<<<< HEAD
```
Comment (chrisong88) [inline react-frontend-app/src/components/Display.js]: Don't forget to resolve this merge conflict on your next commit :)

File: react-frontend-app/src/App.js [hunk 6]
```diff
@@ -12,15 +12,18 @@ import './style/App.css';
...
 import BarGraph from './components/BarGraph'
 
 function App() {
   const [surname, setSurname] = useState('');
   const [zipcode, setZip] = useState('');
-  var prediction = {};
+  const [loading, setLoading] = useState(true);
+
+  // var prediction = {};
+  var [variable, setVariable] = useState('this is the default variable');
```
Comment (chrisong88) [inline react-frontend-app/src/App.js]: I'm really impressed with your command of React Hooks to manage the state variables even within a functional App.js! Great stuff

**Review by chrisong88 — CHANGES_REQUESTED** [review]
Overall, this is great stuff Kayla! Really great work integrating all the different components and adding these new features. There's one major critique that I'd like you to address (it's labeled IMPT) before merging; and then there's also a tiny merge conflict at the bottom.

File: react-frontend-app/src/App.js (hunk 6 above)
Comment (dzheng21) [reply react-frontend-app/src/App.js]: +1 for this!

File: react-frontend-app/src/App.js (hunk 2 above)
Comment (dzheng21) [reply react-frontend-app/src/App.js]: Oh yes, this is a mistake from me — in theory, one would work — @kaylanhua I would just check that it still works if you remove one or the other.

File: react-frontend-app/src/App.js (hunk 3 above)
Comment (dzheng21) [reply react-frontend-app/src/App.js]: This is much cleaner, also another thing in my PR that wasn't great. Ternary operator seems much more clean!

**Review by dzheng21 — APPROVED** [review]
Looks good, seems like major points of change come from my PR :/ lmk if you need my help fixing this stuff, also happy to hop in because these are my mistakes.

**Review by chrisong88 — APPROVED** [review]
Great work on these changes!

**Pull request merged**